docs/infrastructure/benchmarks.md

```diff
@@ -0,0 +1,102 @@
+  measure the performance of specific parts of the engine.
+- [standard library benchmarks](#standard-library-benchmarks) located in the
+  `test/Benchmarks` Enso project. These benchmarks are entirelly written in
+  Enso, along with the harness code.
+
+## Engine JMH microbenchmarks
+
+These benchmarks are written in Java and are used to measure the performance of
+specific parts of the engine. The sources are located in the `runtime` SBT
+project, under `src/bench` source directory.
+
+### Running the benchmarks
+
+To run the benchmarks, use `bench` or `benchOnly` command - `bench` runs all the
+benchmarks and `benchOnly` runs only one benchmark specified with the fully
+qualified name. The parameters for these benchmarks are hard-coded inside the
+JMH annotations in the source files. In order to change, e.g., the number of
+measurement iterations, you need to modify the parameter to the `@Measurement`
+annotation.
+
+### Debugging the benchmarks
+
+Make sure your IDE listens for JDWP connection at port 5005. Debug the benchmark
+by running `withDebug` command like this:
+
+```
+withDebug --debugger benchOnly -- <fully qualified benchmark name>
+```
+
+## Standard library benchmarks
+
+Unlike the Engine micro benchmarks, these benchmarks are written entirelly in
+Enso and located in the `test/Benchmarks` Enso project. There are two ways to
+run these benchmarks:
+
+- [Running standalone](#running-standalone)
+- [Running via JMH launcher](#running-via-jmh-launcher)
+
+### Running standalone
+
+A single source file in the project may contain multiple benchmark definitions.
+If the source file defines `main` method, we can evaluate it the same way as any
+other Enso source file, for example via
+`runEngineDistribution --in-project test/Benchmarks --run <source file>`. The
+harness within the project is not meant for any sophisticated benchmarking, but
+rather for quick local evaluation. See the `Bench.measure` method documentation
+for more details. For more sophisticated approach, run the benchmarks via the
+JMH launcher.
+
+### Running via JMH launcher
+
+The JMH launcher is located in `std-bits/benchmarks` directory, as
+`std-benchmarks` SBT project. It is a single Java class with a `main` method
+that just delegates to the
+[standard JMH launcher](https://github.com/openjdk/jmh/blob/master/jmh-core/src/main/java/org/openjdk/jmh/Main.java),
+therefore, supports all the command line options as the standard launcher. For
+the full options summary, either see the
+[JMH source code](https://github.com/openjdk/jmh/blob/master/jmh-core/src/main/java/org/openjdk/jmh/runner/options/CommandLineOptions.java),
+or run the launcher with `-h` option.
+
+The `std-benchmarks` SBT project supports `bench` and `benchOnly` commands, that
+work the same as in the `runtime` project, with the exception that the benchmark
+name does not have to be specified as a fully qualified name, but as a regular
```

Comment:
Does this not work in `runtime`?

Reply:
(using regexp to filter benchmarks)

Reply:
Not yet, I will look into that in one of the follow-up PRs and unify the functionality even more.